darshan-runtime/lib/darshan-lustre.c

```diff
@@ -228,19 +231,23 @@ static void lustre_runtime_initialize()
         .mod_cleanup_func = &lustre_cleanup
         };
 
+    /* if this attempt at initializing fails, we won't try again */
+    lustre_runtime_init_attempted = 1;
 
-    /* try and store a default number of records for this module, assuming
-     * each file uses 64 OSTs
-     */
-    lustre_buf_size = DARSHAN_DEF_MOD_REC_COUNT * LUSTRE_RECORD_SIZE(64);
+    /* try and store a default number of records for this module */
+    lustre_rec_count = DARSHAN_DEF_MOD_REC_COUNT;
 
     /* register the lustre module with darshan-core */
-    darshan_core_register_module(
+    /* assume each file uses 64 OSTs */
```

Comment:
That's right, the Lustre module needs to provide some indication of how much memory it needs at runtime, but it has dynamic-length records, so just assumes that all records will require 64 OSTs, then requests the default number of Darshan records (1024) with that assumption. The Lustre module has always been coded this way, just that it might be more obvious now given the change to provide both `nrecs` and `sizeof(rec)` to `module_register()`. I'll add comments to make sure this is clear, but you're right that the Lustre module could run out of memory with more or less records than what it requested at init time.